lib/src/http.rs

```diff
@@ -1519,6 +1525,87 @@ impl Proxy {
                 .borrow_mut()
                 .remove_custom_answer(cluster_id);
         }
+        Ok(())
+    }
+
+    pub fn add_http_frontend(&mut self, front: HttpFrontend) -> anyhow::Result<()> {
+        match self
+            .listeners
+            .values()
+            .find(|l| l.borrow().address == front.address)
+        {
+            Some(listener) => {
+                let mut owned = listener.borrow_mut();
+
+                let hostname = front.hostname.to_owned();
+                let tags = front.tags.to_owned();
+
+                match owned.add_http_front(front) {
+                    Ok(_) => {
+                        owned.set_tags(hostname, tags);
+                        Ok(())
+                    }
+                    Err(err) => Err(anyhow::Error::msg(err)),
+                }
+            }
+            None => {
+                // let (listener, tokens) = Listener::new(HttpListener::default(), event_loop,
+                //  self.pool.clone(), None, token: Token) -> (Listener,HashSet<Token>
+                bail!("no HTTP listener found for front: {:?}", front)
+            }
+        }
+    }
+
+    pub fn remove_http_frontend(&mut self, front: HttpFrontend) -> anyhow::Result<()> {
+        match self
```

Comment:
Same here, should be an `if let Some(..) = ..`, if the operation failed because there is no listener the semantic is respected

Reply:
agreed